code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -300,7 +300,11 @@
 			if (sacrifice)
 				walk_to(sacrifice,0)
 		else
-			walk_towards(sacrifice,src, 10)
+			if (istype(sacrifice.loc, /turf) && !(sacrifice.is_ventcrawling) && !(sacrifice.buckled))
+				//Suck the creature towards the pylon if possible
+				walk_towards(sacrifice,src, 10)
+			else
+				walk_to(sacrifice,0) //If we're not in a valid situation, cancel walking to prevent bugginess

```

Comment:
Are you sure you don't mean to call `walk(sacrifice, 0)`? DM ref points to it as being the correct form:
> A call to a walking function aborts any previous walking function called on Ref. **To halt walking, call walk(Ref,0).**

(From walk_to proc definition.)